nhs/data/allocation.py

```diff
@@ -74,6 +74,39 @@ def join_census_with_coords(
     )
 
 
+@log_entry_exit()
+def calculate_proportions(group_values_dict: dict) -> pl.LazyFrame:
+
+    if not group_values_dict:
+        raise ValueError("The grouping dictionary is empty.")
+
+    # Extract the keys (grouping) and values (values) from the dictionary
+    unique_dict = {}
+
+    for group, value in group_values_dict.items():
+        if group not in unique_dict:
+            unique_dict[group] = value
+
+    # Construct a LazyFrame from the unique dictionary
+    lf = pl.LazyFrame(
+        {"group_col": list(unique_dict.keys()), "value_col": list(unique_dict.values())}
+    )
```

Comment:
Could you do this without making another lazy frame? so append a new column onto the existing dataframe for example